wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/convert/DefaultEditorProvisioner.java

```diff
@@ -129,6 +130,18 @@ private String findAvailableName(Set<String> busyNames, String componentRef) {
   }
 
   private String getId(String reference) {
-    return reference.split(":", 2)[0];
+    return resolveIdAndVersion(reference).first;
+  }
+
+  private Pair<String, String> resolveIdAndVersion(String ref) {
```

Comment:
We have the `PluginFQN` class where the parsing results are stored. Maybe it would be useful to introduce a `parse(String)` static method there to centralize the parsing logic. This is the second place where we are parsing the version in a slightly error prone way. The other is, as @sleshchenko metioned, https://github.com/eclipse/che/blob/master/wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/PluginMetaRetriever.java#L147

Reply:
In #12998 I've added a class called [`PluginFQNParser`](https://github.com/eclipse/che/pull/12998/files#diff-4668c3f568aefceb9c6bb2ea3a842259) which could likely handle this.

Reply:
PR is still in draft because relevant changes haven't been released on the plugin-broker side, but apart from that it's pretty much ready for review.

Reply:
@metlos Are you OK with merging this PR and get rid of this duplication in a separate PR when Angel's PR will be merged?

Reply:
:+1: as long as we don't forget to update this usage :smile:

Reply:
Was done in https://github.com/eclipse/che/commit/35ce3544f7e008566daecb1b57266b1bc4fef7f5#diff-440ffdde4e4d35f5eee4e2e9c0644f25R111